### ISSUE TITLE
Docs: Ransomware Protection / Error handling when snapshotting without delete rights

### ISSUE THREAD

**Issue by MurphySunset** [kopia/kopia#4083]
Hello 👋,

I've encountered an issue not covered in the documentation: when creating a snapshot to a remote repository without delete rights, Kopia returns an error due to its inability to remove the session marker. This behavior is expected, but it raises challenges in automation.

#### Example:
```sh
$ kopia snapshot create /home/debian
Snapshotting user@debian:/home/debian ...
Created snapshot with root kb247b55f172a494652ede3e4d67aefd8 and ID 4e159f397bfba41b48f83286b6492be8 in 1s
ERROR flush error: unable to delete session marker sc1050380f4b47cf572ac86ce27c4b0aa-se1812b5e181004eb12c: Access Denied.
```
The command exits with a generic error code:
```sh
$ echo $?
1
```

#### Problem:
While this isn't a major issue when running manually, it complicates error handling in scripts. Users shouldn’t need to parse output to determine if a snapshot was successful.

#### Suggestion:
Consider using a more specific error code for this scenario, allowing scripts to distinguish between generic errors and flushing issues, or please guide me to another solution if this scenario has already been addressed :)

Thanks for considering this improvement or pointing me to a solution!

**Comment by larryclaman** [comment]
There was a bit of a discussion on the forum; does this help?
https://kopia.discourse.group/t/error-failed-to-delete-session-marker/2608/13

**Comment by MurphySunset** [comment]
Issue was coming from the IAM policy which were not that precise in the documentation. It's been addressed: https://github.com/kopia/kopia/pull/3451

tl;dr: DeleteObject needs to be allowed.
